central/declarativeconfig/utils/verify_origin.go

```diff
@@ -0,0 +1,16 @@
+package utils
+
+import (
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/declarativeconfig"
+	"github.com/stackrox/rox/pkg/errox"
+)
+
+func VerifyReferencedResourceOrigin(referencing, referenced declarativeconfig.ResourceWithTraits, referencingName, referencedName string) error {
```

Comment:
We can move this to `pkg/declarativeconfig`, no need to keep it central-specific.